src/payment/strategies/adyenv3/adyenv3.ts

```diff
@@ -0,0 +1,827 @@
+    types?: string[];
+}
+
+export interface PaymentMethodsResponse {
+    /**
+     * Groups of payment methods.
+     */
+    groups?: PaymentMethodGroup[];
+
+    /**
+     * Detailed list of one-click payment methods.
+     */
+    oneClickPaymentMethods?: RecurringDetail;
+
+    /**
+     * Detailed list of payment methods required to generate payment forms.
+     */
+    paymentMethods?: PaymentMethod[];
+
+    /**
+     * List of all stored payment methods.
+     */
+    storedPaymentMethods?: AdyenStoredPaymentMethod[];
+}
+
+export interface RecurringDetail extends PaymentMethod {
+    /**
+     * The reference that uniquely identifies the recurring detail.
+     */
+    recurringDetailReference?: string;
+
+    /**
+     * Contains information on previously stored payment details.
+     */
+    storedDetails?: StoredDetails;
+}
+
+export interface SepaPlaceHolder {
+    ownerName?: string;
+    ibanNumber?: string;
+}
+
+export interface StoredDetails {
+    /**
+     * The stored bank account.
+     */
+    bank?: Bank;
+
+    /**
+     * The stored card information.
+     */
+    card?: Card;
+
+    /**
+     * The email associated with stored payment details.
+     */
+    emailAddress?: string;
+}
+
+export interface StyleOptions {
+    /**
+     * Base styling applied to the iframe. All styling extends from this style.
+     */
+    base?: CssProperties;
+
+    /**
+     * Styling applied when a field fails validation.
+     */
+    error?: CssProperties;
+
+    /**
+     * Styling applied to the field's placeholder values.
+     */
+    placeholder?: CssProperties;
+
+    /**
+     * Styling applied once a field passes validation.
+     */
+    validated?: CssProperties;
+}
+
+export interface SubInputDetail {
+    /**
+     * Configuration parameters for the required input.
+     */
+    configuration?: object;
+
+    /**
+     * In case of a select, the items to choose from.
+     */
+    items?: Item[];
+
+    /**
+     * The value to provide in the result.
+     */
+    key?: string;
+
+    /**
+     * True if this input is optional to provide.
+     */
+    optional?: boolean;
+
+    /**
+     * The type of the required input.
+     */
+    type?: string;
+
+    /**
+     * The value can be pre-filled, if available.
+     */
+    value?: string;
+}
+
+export interface ThreeDS2ChallengeComponentOptions {
+    size?: string;
+    onAdditionalDetails?(state: AdyenAdditionalActionState, component?: AdyenComponent): void;
+    onError(error: AdyenError): void;
+}
+
+export interface ThreeDS2DeviceFingerprintComponentOptions {
+    onAdditionalDetails?(state: AdyenAdditionalActionState, component?: AdyenComponent): void;
+    onError(error: AdyenError): void;
+}
+
+export type AdyenV3ComponentState = (
+    CardState
+);
+
+export type AdyenComponentOptions = (
+    AdyenV3CreditCardComponentOptions | AdyenV3IdealComponentOptions | AdyenCustomCardComponentOptions
+);
+
+export function isCardState(param: any): param is CardState {
```

Comment:
```suggestion
export function isCardState(param: unknown): param is CardState {
```

Reply:
it would not work here.

Reply:
I see but in that case you can possibly use multiple type guards to do the assertion?